c7n/resources/ebs.py

```diff
@@ -50,27 +51,59 @@ class SnapshotAge(AgeFilter):
         days={'type': 'number'},
         op={'type': 'string', 'enum': OPERATORS.keys()})
     date_attribute = 'StartTime'
+    
+
+def _filter_ami_snapshots(self, snapshots):
+# gets a listing of all AMI snapshots and compares resources to the list
+    c = local_session(self.manager.session_factory).client('ec2')
+    ami_snaps = []
+    for i in c.describe_images(Owners=['self'])['Images']:
+        for dev in i.get('BlockDeviceMappings'):
+            if 'Ebs' in dev and 'SnapshotId' in dev['Ebs']:
+                ami_snaps.append(dev['Ebs']['SnapshotId'])
+    matches = []
+    if self.data.get('value', True):
```

Comment:
rather than indent the block, handle False case indented as return snapshots
and then dedent the rest out of this block.